javascript/new-src/src/factories/ola.js

```diff
@@ -22,6 +22,19 @@
 ola.factory('$ola', ['$http', '$window', 'OLA',
   function($http, $window, OLA) {
     'use strict';
+    var highestChannelNumberUsed = 0;
+
+    var updateHighestChannelNumberUsed = function(dmx) {
+      for (var channel = dmx.length; channel > highestChannelNumberUsed;
+           channel--) {
+
+        if ((dmx[channel - 1] > OLA.MIN_CHANNEL_VALUE) &&
+            (highestChannelNumberUsed < channel)) {
+          highestChannelNumberUsed = channel;
+        }
```

Comment:
Would something like this be easier?
```suggestion
        if (dmx[channel - 1] > OLA.MIN_CHANNEL_VALUE) {
          highestChannelNumberUsed = $window.Math.max(highestChannelNumberUsed, channel);
        }
```

Otherwise if you want to be efficient, can't we do some sort of early return, if we're counting down, as we are here, then the first used (non-zero I suspect), channel is what we need, we don't need to check all the following channels.

Edit: Actually I see you already have in the for loop, which I think makes the second part of the `if` redundant doesn't it? If we're looking at this value it's already `channel > highestChannelNumberUsed`.